doc/guides/debugging/probes.rst

```diff
@@ -0,0 +1,143 @@
+.. _debug-probes:
+
+Debug Probes
+############
+
+Many supported boards in Zephyr include a second microcontroller that serves as
+an onboard debug probe, usb-to-serial adapter, and sometimes a drag-and-drop
+flash programmer. This eliminates the need to purchase an external debug probe
+and provides a variety of debug host tool options.
+
+Several hardware vendors have their own branded onboard debug probe
+implementations: NXP LPC boards have LPC-Link2, NXP Kinetis (former Freescale)
+boards have OpenSDA, and ST boards have ST-LINK. Each onboard debug probe
+microcontroller can support one or more types of firmware that communicate with
+their respective debug host tools. For example, an OpenSDA microcontroller can
+be programmed with DAPLink firmware to communicate with pyOCD or openOCD debug
+host tools, or with J-Link firmware to communicate with J-Link debug host
+tools.
+
+Some supported boards in Zephyr do not include an onboard debug probe and
```

Comment:
How about:

```
Boards which do not have integrated debug hardware require an additional external debug
probe to use these features. Usually, these boards have a JTAG or SWD header which can be
connected to the external debug probe via a cable or individual wires.

You can also sometimes use an external debug probe with boards that have integrated debug
hardware. One reason this may be useful is that the integrated hardware may have limitations,
such as lack of support for advanced debuggers or high-speed tracing. You may need to adjust
jumpers on the board to allow debugging via the external probe without interference from the
integrated hardware.
```